vms/rpcchainvm/vm_client.go

```diff
@@ -269,6 +314,17 @@ func (vm *VMClient) BuildBlock() (snowman.Block, error) {
 
 // ParseBlock ...
 func (vm *VMClient) ParseBlock(bytes []byte) (snowman.Block, error) {
+	blkID := hashing.ComputeHash256Array(bytes)
```

Comment:
I don't think this is valid, as we don't assume the hash function used by the VM. In Coreth, the `ID` seems to be generated by `"[Calculating] the keccak256 hash of [the block]'s header."`